fastapi/main.py

```
@@ -0,0 +1,85 @@
+
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S_%f")[:19]  # 밀리초 포함
+    filename  = f"{prefix}_{timestamp}.{ext}"
+    save_path = DEBUG_SAVE_DIR / filename
+
+    save_path.write_bytes(image_bytes)
+    size_kb = len(image_bytes) // 1024
+    print(f"[YOLO DEBUG] 이미지 저장 완료 ─ {save_path.resolve()} ({size_kb}KB)")
+
+
+@app.post("/detect-ingredients")
+async def detect_ingredients(image: UploadFile = File(...)):
```

Ruff (0.15.14)

[warning] Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)